pkg/scheduler/scheduler.go

```diff
@@ -450,6 +471,10 @@ func (s *Scheduler) forwardRequestToQuerier(querier schedulerpb.SchedulerForQuer
 	// Make sure to cancel request at the end to clean up resources.
 	defer s.cancelRequestAndRemoveFromPending(req.FrontendAddress, req.QueryID, "request complete")
 
+	queryComponentName := req.ExpectedQueryComponentName()
+	s.requestQueue.QueryComponentCapacity.IncrementForComponentName(queryComponentName)
+	defer s.requestQueue.QueryComponentCapacity.DecrementForComponentName(queryComponentName)
```

Comment:
For symmetry with the above change (moving the call to `IncrementForComponentName` into the dispatcher loop), what do you think about changing this to a call that is handled by `RequestQueue`'s dispatcher loop as well? 

This would then also mean `QueryComponentCapacity` would no longer have to deal with concurrency at all, as every interaction with it will come from the dispatcher loop.

Reply:
marking down from slack discussion:

the decrementing has to happen in the scheduler because that's the only component that knows when the request finishes.

if we want to both in the request queue we’d have to introduce another channel or similar communication layer to tell the request queue it completed so the queue can call decrement, but that would be out of scope for this PR.